srcStatic/ModuleLoader.cpp

```diff
@@ -74,6 +112,21 @@ std::string ModuleLoader::GetModuleName(std::size_t index)
 	return modules[index]->Name();
 }
 
+std::string ModuleLoader::GetModuleDirectory(std::size_t index)
+{
+	if (index >= modules.size()) {
+		throw std::out_of_range("GetModuleDirectory: Invalid console module index: " + std::to_string(index));
+	}
+
+	ConsoleModule* consoleModule = dynamic_cast<ConsoleModule*>(modules[index].get());
+	if (consoleModule != nullptr) {
+		return consoleModule->Directory();
```

Comment:
This is correct, though I might prefer is the `dynamic_cast` was not required. We could get rid of it if the base class guaranteed to provide a `Directory()` method. The base class could have a default implementation that returns the empty string.